include/boost/http_proto/serializer.hpp

```diff
@@ -192,6 +191,45 @@ class BOOST_SYMBOL_VISIBLE
     make_array(std::size_t n) ->
         detail::array_of_const_buffers;
 
+    template<
+        class Source,
+        class... Args,
+        typename std::enable_if<
+            std::is_constructible<
+                Source,
+                Args...>::value>::type* = nullptr>
+    Source&
+    construct_source(Args&&... args)
+    {
+        // TODO: add ws_.emplace<T>(...) interface
+        // to prevent the unnecessary move
+        return ws_.push(
+            Source(std::forward<Args>(args)...));
+    }
+
+    template<
+        class Source,
+        class... Args,
+        typename std::enable_if<
+            std::is_constructible<
+                Source,
+                buffered_base::allocator&,
+                Args...>::value>::type* = nullptr>
+    Source&
+    construct_source(Args&&... args)
```

Comment:
You need to somehow static_assert if Source is not constructible either way, or else an ugly compile error results.